src/cmd/fix.rs

```diff
@@ -388,11 +504,51 @@ async fn check_integrations<L: Logger>(logger: &L, config: &Config) -> Vec<Error
 }
 
 #[tracing::instrument]
-async fn check_integration(integration: &Integration) -> Result<(), Error> {
+async fn check_integration_connection(integration: &Integration) -> Result<(), Error> {
     let Protocol::Git(transport) = integration.protocol();
-    repository::ls_remote(transport)
+    repository::ls_remote(transport).await.or_else(|err| {
+        Error::integration_connection_error(integration.remote(), transport, err).wrap_err()
+    })?;
+    Ok(())
+}
+
+#[tracing::instrument]
+async fn check_integration_scan(
+    ctx: &AppContext,
+    config: &Config,
+    integration: &Integration,
+) -> Result<(), Error> {
+    let remote = integration.remote();
+    let cli = fossa_cli::find_or_download(ctx, config.debug().location(), DesiredVersion::Latest)
         .await
-        .or_else(|err| Error::integration_error(integration.remote(), transport, err).wrap_err())?;
+        .or_else(|_err| Error::download_cli_error(remote).wrap_err())?;
+
+    let references = integration.references().await.unwrap_or_default();
+    if references.is_empty() {
+        return Error::EmptyReferences.wrap_err();
+    }
+    let mut reference = &references.clone()[0];
+
+    let main_reference = references
+        .into_iter()
+        .filter(|r| r.get_name().as_str() == MAIN_BRANCH || r.get_name().as_str() == MASTER_BRANCH)
+        .collect::<Vec<_>>();
+
+    if !main_reference.is_empty() {
+        reference = &main_reference[0];
+    }
```

Comment:
Searching to scan main/master branch. If that isn't present take the first reference in the list of references. 

I had a previous conversation with @jssblck regarding empty references at this stage. Is this a realistic possibility and is the fix to just point the user to the broker readme for proper setup instructions?